zerver/lib/markdown/fenced_code.py

```diff
@@ -472,8 +473,19 @@ def format_quote(self, text: str) -> str:
         for paragraph in paragraphs:
             lines = paragraph.split("\n")
             quoted_paragraphs.append("\n".join("> " + line for line in lines))
+            quoted_paragraphs = [self.format_quoted_emoji(quote) for quote in quoted_paragraphs]
         return "\n".join(quoted_paragraphs)
 
+    # Used when quoting a message with an emoticon
+    # to prevent its translation to an emoji
+    # and preserve the original message context
+    def format_quoted_emoji(self, text: str) -> str:
+        for m in re.finditer(emoticon_regex, text):
+            symbol_re = re.compile(re.escape(m.group("emoticon")))
```

Comment:
This will work even without using `re.compile`; we should use `re.compile` when that `symbol_re` is used at multiple places.